File: GraphPaper_VB/Globals.vb
Imports System.ComponentModel

Module Globals
    Public Settings As SharkInSeine.Settings
    Public DesignMode As Boolean = LicenseManager.UsageMode = LicenseUsageMode.Designtime
    Public Sub WriteSettings()
        If Settings IsNot Nothing AndAlso Not DesignMode Then
            Settings.Write()
        End If
    End Sub
    Public Property UsePrintMargins
        Get
            If Settings Is Nothing OrElse DesignMode Then
                Return True
            Else
                Return Settings.GetValueBoolean("Main", "UsePrintMargins", True)
            End If
        End Get
        Set(value)
            If value <> UsePrintMargins Then
                If Settings IsNot Nothing Then
                    Settings.SetValue("Main", "UsePrintMargins", value)
                    WriteSettings()
                End If
            End If
        End Set
    End Property
End Module

File: GraphPaperCsharp/Globals.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SharkInSeine
{
    internal class Globals
    {
        static public Settings.Settings Settings;
        static public bool DesignMode = LicenseManager.UsageMode == LicenseUsageMode.Designtime;
        public static void WriteSettings()
        {
            if (Settings != null && !DesignMode)
            {
                Settings.Write();
            }
        }
        static public bool UsePrintMargins
        {
            get
            {
                if (Settings== null || DesignMode)
                {
                    return true;
                }
                else
                {
                    return Settings.GetValueBoolean("Main", "UsePrintMargins", true);
                }
            }
            set
            {
                if (value != UsePrintMargins)
                {
                    if (Settings != null)
                    {
                        Settings.SetValue("Main", "UsePrintMargins", value);
                        Settings.Write();
                    }
                }
            }
        }
    }
}
